qcfractal/queue/managers.py

```diff
@@ -309,29 +336,105 @@ def add_exit_callback(self, callback: Callable, *args: List[Any], **kwargs: Dict
         """
         self.exit_callbacks.append((callback, args, kwargs))
 
-    def update(self, new_tasks: bool=True) -> bool:
+    def _post_update(self, payload_data, allow_shutdown=True):
+        """Internal function to post payload update"""
+        payload = self._payload_template()
+        # Update with data
+        payload["data"] = payload_data
+        try:
+            self.client._automodel_request("queue_manager", "post", payload)
+        except IOError:
+            # Trapped behavior elsewhere
+            raise
+        except Exception as fatal:
+            # Non IOError, something has gone very wrong
+            self.logger.error("An error was detected which was not an expected requests-type error. The manager "
+                              "will attempt shutdown as best it can. Please report this error to the QCFractal "
+                              "developers as this block should not be "
+                              "seen outside of debugging modes. Error is as follows\n{}".format(fatal))
+
+            try:
+                if allow_shutdown:
+                    self.shutdown()
+            finally:
+                raise fatal
+
+    def _update_stale_jobs(self, allow_shutdown=True):
+        """
+        Attempt to post the previous payload failures
+        """
+        clear_indices = []
+        for index, (results, attempts) in enumerate(self._stale_payload_tracking):
+            try:
+                self._post_update(results)
+                self.logger.info(f"Successfully pushed jobs from {attempts+1} updates ago")
+                clear_indices.append(index)
+            except IOError:
+                # Tried and failed
+                attempts += 1
+                # Case: Still within the retry limit
+                if self.server_error_retries is None or self.server_error_retries > attempts:
+                    self._stale_payload_tracking[index][-1] = attempts
+                    self.logger.warning(f"Re-attempt to post jobs from {attempts} ago failed again, will retry")
```

Comment:
Wording is a little odd if the fstring evals. `Could not post jobs, {attempts} previous attempts`?

Reply:
How does `Could not post jobs from {attempts} ago, will retry on next update.` sound?